src/draw/lv_image_decoder.h

```diff
@@ -76,6 +76,15 @@ typedef lv_result_t (*lv_image_decoder_get_area_cb_t)(lv_image_decoder_t * decod
  */
 typedef void (*lv_image_decoder_close_f_t)(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc);
 
+/**
+ * Custom drawing functions for special image formats.
+ * @param layer pointer to a layer
+ * @param dsc pointer to decoder descriptor
+ * @param coords the coordinates of the image
+ * @param draw_dsc the draw image descriptor
+ */
+typedef void (*lv_image_decoder_custom_draw_t)(lv_layer_t * layer, const lv_image_decoder_dsc_t * dsc,
+                                               const lv_area_t * coords, const lv_draw_image_dsc_t * draw_dsc);
```

Comment:
What about introducing a `LV_IMAGE_FLAGS_VECTOR` in `lv_image_flags_t` and setting it in `decoder_open`. This way if a draw unit can't support vector images doesn't take the image draw task. And if it supports it, the image can be drawn as it wants. 

I think it would be cleaner as it fits into the current rendering flow.

Reply:
This provides a general mechanism for decoders, not just SVG images. In the future, all content that needs a custom drawing method can use this interface to implement a custom drawing mechanism, which is a more general solution.

Reply:
I feel like it's to custom as it ignores the whole draw task and draw unit concept:
- the biggest problem: the image is drawn immediately. Normally draw_tasks are added to a list and consumed from there. With this change instead of adding the draw task to the end of the list, it will be rendered immediately.
- draw units are used out of order, which is not thread safe
- draw task is not created, no `LV_EVENT_DRAW_TASK_ADDED` is emitted in `lv_draw_finalize_task_creation`

Reply:
In fact, this will not happen, because drawing with the vector api will eventually call the `lv_draw_vector` function, creating the correct draw_task task, but the type becomes `LV_DRAW_TASK_TYPE_VECTOR`

Reply:
True, it's a very good point.